### ISSUE TITLE
Functions with no arguments

### ISSUE THREAD

**Issue by ViliamVadocz** [LHolten/Volpe#38]
As of commit cc70997 functions without arguments are not possible:
```
get_1 := () {1}
```
These types of functions are useful if there is some computation that could be carried out, but it doesn't always have to be:
```
vec := (x, y, z) {
    vec := @
    add := (b) { vec(x + b.x, y + b.y, z + b.z)
    dot := (b) { x * b.x + y * b.y + z * b.z }
    norm := () { sqrt(x * x + y * y + z * z }  # we don't always need to compute norm
    x, y, z, add, dot, norm
}
```

As a side note, empty objects are also not possible: `{}` (they're not very useful anyway)

**Comment by LHolten** [comment]
Instead of adding back functions without arguments we might make object attributes lazy

**Comment by LHolten** [comment]
this has been resolved in bb8089592bbf019a536d5e6fdf281d1ee5561075
